cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -210,25 +241,45 @@ runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
     CardanoMode -> do
       let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
 
-      (chainTip, eLocalState) <- liftIO $
-        executeLocalStateQueryExprWithChainSync localNodeConnInfo Nothing $ \ntcVersion -> do
-          era <- queryExpr (QueryCurrentEra CardanoModeIsMultiEra)
-          eraHistory <- queryExpr (QueryEraHistory CardanoModeIsMultiEra)
-          mSystemStart <- if ntcVersion >= NodeToClientV_9
-            then Just <$> queryExpr QuerySystemStart
-            else return Nothing
-          return O.QueryTipLocalState
-            { O.era = era
-            , O.eraHistory = eraHistory
-            , O.mSystemStart = mSystemStart
-            }
+      eLocalState <- liftIO $ executeLocalStateQueryExpr localNodeConnInfo Nothing $ \ntcVersion -> do
+        era <- queryExpr (QueryCurrentEra CardanoModeIsMultiEra)
+        eraHistory <- queryExpr (QueryEraHistory CardanoModeIsMultiEra)
+        mChainBlockNo <- if ntcVersion >= NodeToClientV_10
```

Comment:
We really need to come up with a better way of handling which queries are available for a particular version. Is this information available in the consensus library?

Reply:
At the moment, it's document in the code here: https://github.com/input-output-hk/ouroboros-network/blob/c25d36e62549d5bec9876cb57d1c52ee41bdec3a/ouroboros-network/src/Ouroboros/Network/NodeToClient/Version.hs#L45